src/content/docs/ai-monitoring/configure-ai-monitoring.mdx

```diff
@@ -0,0 +1,155 @@
+---
+title: 'Configure AI monitoring'
+metaDescription: 'You can apply certain configurations to your APM agents to change how your AI data appears in New Relic.'
+freshnessValidatedDate: never
+---
+
+Once you install AI monitoring, you can configure the default behavior of the agent or update your app to collect different kinds of data. 
+
+## Configure the agent [#configure-agents]
+
+Update default agent behavior for AI monitoring at these agent configuration docs: 
+
+* [Go](/docs/apm/agents/go-agent/configuration/go-agent-configuration/#ai-monitoring)
+* [Node.js](/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration/#ai-monitoring)
+* [Python](/docs/apm/agents/python-agent/configuration/python-agent-configuration/#ai-monitoring)
+* [Ruby](/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#ai-monitoring)
+
+## Enable token API [#enable-token]
+
+If you've opted to drop message contents from LLM event data, you can still forward token count information to New Relic without storing message contents from your event data. If you haven't disabled message contents from being stored in NRDB, you don't need to enable the token API. 
+
+Update your app code with the below code snippets: 
+
+<CollapserGroup>
+    <Collapser
+        id="nodejs-token-api"
+        title="Node.js"
+    >
+
+    ```js
+    // register callback to set token counts 
+      newrelic.setLlmTokenCount(callback)
+
+      /**
+      * Sample callback to calculate token count. Must be a synchronous method.
+      * This callback is optional, and is designed for customers that cannot send
+      * the content in the Llm events but still want token counts in the case
+      * they are not present.
+      * @param {string} model name of model
+      * @param {string} content content of call
+      * @returns {int} token count for content, customers must use relevant libraries to calculate this
+      */
+      function callback(model, content) {
```

Comment:
this example is bare bones. I was hoping @bonczj could provide more insights into properly tokenizing content as he helped drive this in the pipeline

Reply:
@natecanfield822 i agree with @bizob2828 that the more in depth / fleshed out example we have, the better. having in depth procedural content about how/when/where to apply these in their environment would be very helpful too. we can cut away the fat as we draft, but for these docs i think more verbose is better

Reply:
I'd ensure we add a note that mentions this is an _example_ and not the only way to do it (whether we keep this example or add a more complex one)

Reply:
Fair points. I think part of our challenge is going to be availability across languages - while it looks like [tiktoken](https://www.npmjs.com/package/tiktoken) has a node release for example, and definitely could be used as the example for python, there's no official Ruby release - as far as I can tell we'd have to essentially pick a random third party unofficial tokenizer for that example, which feels a bit iffy to put in our docs.

Reply:
I will review the full text today/tomorrow to make sure we are accurately documenting the process and showing examples as illustration purposes only. I think it will be fairly easy to show examples of calculating token counts for Python and Node for some models, with a goal of trying to focus on one.

As an example, for Node even OpenAI [recommends](https://platform.openai.com/tokenizer) a specific community source [package](https://www.npmjs.com/package/tiktoken) though OpenAI has released `tiktoken` for Python under their own github account.

Reply:
Node example.

```js
const tiktoken = require('tiktoken')

newrelic.setLlmTokenCountCallback(tokenCountCallback)

/**
 * Calculate token counts locally based on the model being used and the content.
 * This callback will be invoked for each message as part of a communication with the LLM.
 * If the application supports more than one model, it may require finding libraries for
 * each model to support token counts appropriately.
 * 
 * @param {string} model - name of model (i.e. gpt-3.5-turbo)
 * @param {string} content - prompt or completion response
 * @returns {number} number of tokens that represent the supplied content for the given model
 */
function tokenCountCallback(model, content) {
  if (model.includes('gpt')) {
    const enc = tiktoken.encoding_for_model(model)
    const count = enc.encoding(content).length()
    enc.free()

    return count
  }

  // for models that are not supported in the example
  return 0
}
```

Reply:
We might want to make this point to each of the agent's individual API docs as opposed to trying to make multiple examples here for each agent. Certainly for Python we will be adding a new public API for this with it's own examples (under `src/content/docs/apm/agents/python-agent/python-agent-api/`). Looks like Node has a similar area in their docs as well: `src/content/docs/apm/agents/nodejs-agent/api-guides/`. We were originally going to make a separate PR for this but it might be better to just add it into this one so you can point at it in these docs. Same for the record feedback event and current_trace_id. What do you all think?